src/domain/community/user/user.resolver.queries.ts

```diff
@@ -64,38 +61,24 @@ export class UserResolverQueries {
   }
 
   @UseGuards(GraphqlGuard)
-  @Query(() => RelayStylePaginatedUser, {
+  @Query(() => PaginatedUsers, {
     nullable: false,
     description: 'The users who have profiles on this platform',
   })
   @Profiling.api
   async usersPaginated(
     @CurrentUser() agentInfo: AgentInfo,
-    @Args({
-      name: 'first',
-      type: () => Int,
-      description: 'Amount of users after the cursor',
-      nullable: true,
-    })
-    first: number,
-    @Args({
-      name: 'after',
-      type: () => UUID,
-      description: 'Selects the items after the cursor',
-      nullable: true,
-    })
-    after: string
-  ): Promise<RelayStylePaginatedUser> {
+    @Args() pagination: PaginationArgs,
+    @Args() filter: FilterArgs
+  ): Promise<PaginatedUsers> {
     await this.authorizationService.grantAccessOrFail(
       agentInfo,
       this.authorizationPolicyService.getPlatformAuthorizationPolicy(),
       AuthorizationPrivilege.READ_USERS,
       `users query: ${agentInfo.email}`
     );
 
-    // todo: use different method without offset based pagination
-    const list = await this.userService.getUsers();
-    return getRelayStylePaginationResults(list, first, after);
+    return this.userService.getPaginatedUsers(pagination, filter);
```

Comment:
Not awaited?

Reply:
No need. We do not need the value here - it's just returned.